### PR TITLE
WIP: Line diagram changes for OL Surge

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes
**Asana Ticket:** [Prototype line diagram changes](https://app.asana.com/0/555089885850811/1202807031649874/f)

The Orange line page will remain unchanged - but if you include a `date_time` parameter it'll get picked up - use **`/schedules/Orange/line?date_time=2022-08-20T12:22`** to see the page during surge.

Initially looking something like this - feedback wanted on more content! We need details on connections I'd think.

image

Builds on #1381 (first two commits).

Putting up this PR before I finish writing enough tests to pass coverage - will add them here.

---

Before getting review, please check the following:

* ~[ ] Does frontend functionality render and work correctly in IE?~
* [ ] Have we load-tested any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md) (n/a)
* [ ] Are interactive elements accessible to screen readers? (hasn't changed)
* [x] Have you checked for tech debt you can address in the area you're working in?
* [x] If this change involves routes, does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?
* [x] Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?

### CODE REVIEW

**Review by zovt — COMMENTED** [review]
Cool stuff so far!

File: apps/site/assets/ts/helpers/use-line-suspensions.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+interface SuspendedStopsInfo {
+  [routePatternId: string]: SuspendedStopsList;
+}
+
+interface SuspendedStopsList {
+  [directionId: string]: string[];
+}
+
+const currentLineSuspensions = (routeId: string): LineSuspensionInfo => {
+  const info: LineSuspensionInfo = {
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/helpers/use-line-suspensions.ts:32]: Yeah, as a backup return value if there are no suspensions found. Not sure if that's necessarily better than returning `null` or `undefined` but I chose it so I wouldn't have to add `| null` to the return signature of this function or check for null everywhere I use this output.

File: apps/site/assets/ts/helpers/use-line-suspensions.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+  if (!document) return info;
+
+  const lineSuspensionsEl: Element | undefined = document.querySelector(
+    "[data-line-suspensions]"
+  )!;
+  if (!lineSuspensionsEl) return info;
+
+  const suspensions = parseLineSuspensions(
+    JSON.parse(lineSuspensionsEl.innerHTML)
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/helpers/use-line-suspensions.ts:46]: Marvelous, thanks!

File: apps/site/assets/ts/helpers/use-line-suspensions.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+const parseLineSuspensions = (
+  suspensionsList: [string, TimePeriodPairs][]
+): LineSuspensions =>
+  suspensionsList.reduce((acc, [routeId, dateRange]) => {
+    acc[routeId] = dateRange;
+    return acc;
+  }, {} as LineSuspensions);
+
+interface LineSuspensionInfo {
+  lineIsSuspended: boolean;
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/helpers/use-line-suspensions.ts]: Good points - I refactored this and ended up returning `LineSuspensionInfo | undefined`

File: apps/site/assets/ts/helpers/__tests__/use-line-suspensions-test.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import currentLineSuspensions from "../line-suspensions";
+
+// TODO: Add tests
```
Comment (zovt) [inline apps/site/assets/ts/helpers/__tests__/use-line-suspensions-test.ts:3]: not blocking but could be worth a follow up ticket to clean up the TODOs around here if we're trying to push this out today

**Review by zovt — APPROVED** [review]
I think this all checks out to me 🚋

**Comment by thecristen** [comment]
Sorry, just one more change! ✨ Adds recommended connections

image

**Review by zovt — APPROVED** [review]
🔌

**Pull request merged**